topics/data-science/tutorials/python-loops/tutorial.md

```diff
@@ -0,0 +1,551 @@
+---
+layout: tutorial_hands_on
+
+title: Python - Loops
+level: Introductory
+requirements:
+ - type: internal
+   topic_name: data-science
+   tutorials:
+     - python-iterables
+     - python-flow
+follow_up_training: []
+questions:
+- "How can I make a program do many things?"
+
+objectives:
+- "Explain what for loops are normally used for."
+- "Trace the execution of a simple (unnested) loop and correctly state the values of variables in each iteration."
+- "Write for loops that use the Accumulator pattern to aggregate values."
+
+time_estimation:  40M
+key_points:
+- "A *for loop* executes commands once for each value in a collection."
+- "A `for` loop is made up of a collection, a loop variable, and a body."
+- "The first line of the `for` loop must end with a colon, and the body must be indented."
+- "Indentation is always meaningful in Python."
+- "Loop variables can be called anything (but it is strongly advised to have a meaningful name to the looping variable)."
+- "The body of a loop can contain many statements."
+- "Use `range` to iterate over a sequence of numbers."
+- "The Accumulator pattern turns many values into one."
+
+enable: false
+subtopic: python-modular
+contributors:
+  - carpentries
+  - hexylena
+  - dirowa
+
+priority: 1
+notebook:
+  language: python
+---
+
+A *for loop* tells Python to execute some statements once for each value in a list, a character string, or some other collection: "for each thing in this group, do these operations"
+
+> ### {% icon comment %} Comment
+> This tutorial is **significantly** based on [the Carpentries](https://carpentries.org) [Programming with Python](https://swcarpentry.github.io/python-novice-inflammation/), [Programming with Python](https://swcarpentry.github.io/python-novice-inflammation/), and [Plotting and Programming in Python](https://swcarpentry.github.io/python-novice-gapminder/), which are licensed CC-BY 4.0.
+>
+> Adaptations have been made to make this work better in a GTN/Galaxy environment.
+{: .comment}
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# For Loops
+
+Which of these would you rather write
+
+> > ### {% icon code-in %} Manually
+> > ```
+> > print(2)
+> > print(3)
+> > print(5)
+> > print(7)
+> > print(11)
+> > ```
+> {: .code-in}
+>
+> > ### {% icon code-out %} With Loops
+> > ```python
+> > for number in [2, 3, 5, 7, 11]:
+> >     print(number)
+> > ```
+> {: .code-out}
+{: .code-2col}
+
+It may be less clear here, since you just need to do one operation (`print`) but if you had to do two operations, three, more?
+
+## Structure
+
+## A `for` loop is made up of a collection, a loop variable, and a body.
+
+```python
+for number in [2, 3, 5]:
+    doubled = number * 2
+    print(f"{number} doubled is {doubled}")
```

Comment:
for some reason does the print function stay out of the loop in CC. maybe a small bug?